internal/util/util.go

```diff
@@ -19,19 +19,96 @@ package util
 import (
 	"crypto/sha1"
 	"fmt"
+	"sort"
 
-	"helm.sh/helm/v3/pkg/chartutil"
+	goyaml "gopkg.in/yaml.v2"
 	"helm.sh/helm/v3/pkg/release"
+	"sigs.k8s.io/yaml"
 )
 
 // ValuesChecksum calculates and returns the SHA1 checksum for the
 // given chartutil.Values.
-func ValuesChecksum(values chartutil.Values) string {
-	var s string
-	if len(values) != 0 {
-		s, _ = values.YAML()
+func ValuesChecksum(values map[string]interface{}) string {
+	newValues := copyValues(values)
+	var (
+		s   []byte
+		err error
+	)
+
+	if len(newValues) != 0 {
+		// cleanUpInterfaceMap
+		for i, value := range newValues {
+			newValues[i] = cleanUpMapValue(value)
+		}
+		msValues := yaml.JSONObjectToYAMLObject(newValues)
+		// Sort
+		SortMapSlice(msValues)
+		// Marshal
+		s, err = goyaml.Marshal(msValues)
+		if err != nil {
+			panic(err)
```

Comment:
Panics based on arbitrary input with little guarantee it may only occur due to programmatic errors (and not based on user-input) make me feel uncomfortable.